src/logic/messages/EventManager.h

```diff
@@ -18,7 +18,7 @@ namespace Logic
         ~EventManager();
 
         template<typename T>
-        void onMessage(const T& msg, Handle::EntityHandle sourceVob = Handle::EntityHandle::makeInvalidHandle())
+        T* onMessage(const T& msg, Handle::EntityHandle sourceVob = Handle::EntityHandle::makeInvalidHandle())
```

Comment:
Please, don't return a raw pointer allocated in this function, people tend to forget about freeing the memory afterwards, use unique_ptr instead.